cm/master.py

```diff
@@ -259,11 +259,11 @@ def load_legacy_snapshot_data(self):
         if s3_conn and misc.get_file_from_bucket(s3_conn, self.app.config['bucket_default'],
            'snaps.yaml', snaps_file, validate=validate):
             pass
-        elif misc.get_file_from_public_bucket(self.app.config, self.app.config['bucket_default'],
-                                              'snaps.yaml', snaps_file):
+        elif misc.get_file_from_public_location(self.app.config, 'snaps.yaml', snaps_file):
             log.warn("Couldn't get snaps.yaml from bucket: {0}. However, managed "
-                     "to retrieve it from public S3 bucket {0} instead."
-                     .format(self.app.config['bucket_default']))
+                     "to retrieve it from public location '{1}' instead.".format(
+                     self.app.config['bucket_default'],
+                     self.app.config.get('default_bucket_url') or self.app.config['bucket_default']))
```

Comment:
I do agree with what you've been talking about. Bucket_default is applicable cloudman-wide, and is used often too, and therefore, is a good candidate for the global config. For the moment, we should probably leave this as is as Simon suggested, and do a global search replace during a future refactoring run.

This is a really useful pull-request Simon and should prove ideal, since there's a suggestion to eliminate the default bucket altogether, and move to a url based approach to bootstrapping cloudman.